### PR TITLE
Working version

### PR DESCRIPTION
**Pull request by doudou**
This is a big code drop ... Should have pushed things more little-by-little.

In any case, this works in the sense that:
- we have access to all the configuration parameters I think we will need
- we have access to almost all the data we will need (needed to add covariances for instance)
   > we're missing calculation of the euler angle covariances (see https://forum.aceinna.com/topic/58/convert-covariance-of-quaternions-into-covariance-of-euler-angles)
- I have a good idea of the development process for the firmware (using VSCode and PlatformIO, so far I don't think we should spend effort making the build process headless)

### CODE REVIEW

**Comment by g-arjones** [comment]
I'm wondering if there is a reason for the metaprogramming in the endianness conversion functions...

```cpp
template<typename T>
uint8_t const* decode(uint8_t const* buffer, T& value)
{
    T result = 0;
    uint8_t shifter = 0;
    for (size_t i = 0; i < sizeof(T); i++) {
        result |= static_cast<T>(buffer[i]) << shifter;
        shifter += 8;
    }
    value = reinterpret_cast<T const&>(result);
    return buffer + sizeof(T);
}


template<typename T>
uint8_t* encode(uint8_t* buffer, T const& value)
{
    uint8_t shifter = 0;
    T bytes = reinterpret_cast<T const&>(value);
    for (size_t i = 0; i < sizeof(T); i++) {
        buffer[i] = (bytes >> shifter) & 0xFF;
        shifter += 8;
    }
    return buffer + sizeof(T);
}
```

What am I missing?

**Comment by doudou** [comment]
> What am I missing?

Not much, really. We'll just need to compute the integer-type-with-the-right-size but other than that your version is probably going to be optimized into something that closely looks what I wrote by hand.

I'd stay with my version for now, though, if you don't mind. Just so that I can focus on the goal at hand and maybe come back to it another time ?

File: src/EKFWithCovariance.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#include <imu_aceinna_openimu/FilterState.hpp>
+
+namespace imu_aceinna_openimu {
+    /** The firmware's EKF-with-covariance message (e3) */
+    struct EKFWithCovariance {
+        base::samples::RigidBodyState rbs;
+        base::samples::RigidBodyAcceleration rba;
+        FilterState filter_state;
+
+        double latitude = base::unknown<double>();
```
[the remark this answers is not in the document]
Comment (doudou) [reply src/EKFWithCovariance.hpp]: Mmmm. Yes. I'm missing these since it's enabled by default on 18.04-and-later. Added it.

**Comment by g-arjones** [comment]
> if you don't mind

Absolutely, go ahead... :+1:

**Pull request merged**